source/extensions/filters/http/grpc_json_transcoder/json_transcoder_filter.cc

```diff
@@ -887,6 +900,50 @@ bool JsonTranscoderFilter::maybeConvertGrpcStatus(Grpc::Status::GrpcStatus grpc_
   return true;
 }
 
+bool JsonTranscoderFilter::decoderBufferLimitReached(uint64_t buffer_length) {
+  if (!Runtime::runtimeFeatureEnabled(buffer_limits_runtime_feature)) {
+    return false;
+  }
+
+  if (buffer_length > decoder_callbacks_->decoderBufferLimit()) {
+    ENVOY_LOG(debug,
+              "Request rejected because the transcoder's internal buffer size exceeds the "
+              "configured limit: {} > {}",
+              buffer_length, decoder_callbacks_->decoderBufferLimit());
+    error_ = true;
+    decoder_callbacks_->sendLocalReply(
+        Http::Code::PayloadTooLarge,
+        "Request rejected because the transcoder's internal buffer size exceeds the configured "
+        "limit.",
+        nullptr, absl::nullopt,
+        absl::StrCat(RcDetails::get().GrpcTranscodeFailed, "{request_buffer_size_limit_reached}"));
```

Comment:
Why is the string in braces? Same below?

Reply:
This follows the pre-existing format for Response Code Details. Other examples from across Envoy:
```
upstream_reset_before_response_started{remote_reset}
jwt_authn_access_denied{Jwt_is_missing}
grpc_json_transcode_failure{INVALID_ARGUMENT}
```

Reply:
I'm not sure why the others are like that. I don't think the braces here add anything. Can they be removed?

/wait

Reply:
Hmm, I'd prefer not to change the existing format in this PR. This filter also has other `sendLocalReply` with the braces. Should we do this in a follow-up PR?

Reply:
We can leave the existing ones but I don't see why we would add new ones like this.

/wait

Reply:
Ack, done.